CONTRIBUTING.md

```diff
@@ -1,94 +1,354 @@
+
+To add a community pipeline, one should add a <name-of-the-community>.py file to [examples/community](https://github.com/huggingface/diffusers/tree/main/examples/community) and adapt the [examples/community/README.md](https://github.com/huggingface/diffusers/tree/main/examples/community/README.md) to include an example of the new pipeline.
+
+TODO(PVP:Copy&Paste test from the issue)
+
+Community pipeline PRs are only checked at a superficial level and it is desired to be maintained by the author.
+
+Contributing a community pipeline is a great way to understand how Diffusers models and schedulers work. Having contributed a community pipeline is usually the first stepping stone to contributing an official pipeline to the
+core package.
+
+### 7. Contribute to training examples
+
+Diffusers examples are a collection of training scripts that reside in [examples](https://github.com/huggingface/diffusers/tree/main/examples).
+
+We support two types of training examples:
+
+- Official training examples
+- Research training examples
+
+Research training examples are located in [examples/research_projects](https://github.com/huggingface/diffusers/tree/main/examples/research_projects) whereas official training examples include all folders under [examples](https://github.com/huggingface/diffusers/tree/main/examples) except the `research_projects` and `community` folders.
+The official training examples are maintained by the Diffusers core-maintainers whereas the research training examples are maintained by the community.
+This is because of the same reasons put forward in [6. Contribute a community pipeline](#contribute-a-community-pipeline) for official pipelines vs. community pipelines: It is not feasible for the core maintainers to maintain all possible training methods for diffusion models.
+If the Diffusers core-maintainers and community judges a certain training paradigm is not popular enough or too experimental, the corresponding training code should be put in the `research_projects` folder and maintained by the author.
+
+Both official training and research examples consist of a directory that contains one or more training scripts, a requirements.txt file, and a README.md file. In order for the user to make use of the
+training examples, it is required to git clone the repository:
+
+```
+git clone https://github.com/huggingface/diffusers
+```
+
+as well as to install all additional dependencies required for training:
+
+```
+pip install -r /examples/<your-example-folder>/requirements.txt
+```
+
+Therefore when adding an example, the requirements.txt file shall define all pip dependencies required for your training example so that once all those are installed the user can run the example's training script, as an example, see the [dreambooth requirements.txt file](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/requirements.txt).
```

Comment:
```suggestion
Therefore when adding an example, the `requirements.txt` file shall define all pip dependencies required for your training example so that once all those are installed, the user can run the example's training script, as an example, see the [DreamBooth requirements.txt file](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/requirements.txt).
```